### PR TITLE
Write reminders to DB

### PR DESCRIPTION
**Pull request by tomrf1**
also add custom validation for email and date in the client request

### CODE REVIEW

File: src/lib/log.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- ignore
+export function logWarning(msg: string, ...args: any[]): void {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- ignore
+	console.log.apply(null, [`[WARN] ${msg}`, ...args]);
+}
+// eslint-disable-next-line @typescript-eslint/no-explicit-any -- ignore
+export function logInfo(msg: string, ...args: any[]): void {
+	// eslint-disable-next-line @typescript-eslint/no-unsafe-assignment -- ignore
+	console.log.apply(null, [`[INFO] ${msg}`, ...args]);
+}
```
Comment (TomPretty) [inline src/lib/log.ts:15]: That's a good ratio of code to eslint ignores 😂

File: src/create-reminder-signup/lambda/lambda.ts [hunk 2]
```diff
@@ -1,65 +1,79 @@
...
+const dbConnectionPoolPromise: Promise<Pool> = getDatabaseParamsFromSSM(
+	ssm,
+).then(createDatabaseConnectionPool);
 const identityAccessTokenPromise: Promise<string> = getParamFromSSM(
 	ssm,
 	`/support-reminders/idapi/${ssmStage}/accessToken`,
 );
 
-export const handler = (
+export const handler = async (
```
Comment (paulbrown1982) [inline src/create-reminder-signup/lambda/lambda.ts:30]: I remember there was a team which had an issue with async handlers for node.js lambdas...

Comment (tomrf1) [reply src/create-reminder-signup/lambda/lambda.ts:30]: haha yes, that was us. Curiously the aws advice was not to do it until you hit a problem, but we don't want to lose data so we should really just do it now.
I had to do it for referrals here: https://github.com/guardian/contributions-referrals/pull/26

Comment (tomrf1) [reply src/create-reminder-signup/lambda/lambda.ts:30]: i'll do this in a follow up PR - I think it's potentially an issue regardless of the `async` keyword

File: src/create-reminder-signup/lambda/models.ts [hunk 3]
```diff
@@ -1,25 +1,44 @@
...
+	reminder_option?: string;
+}
 
-export type ReminderComponent =
-	'EPIC' |
-	'BANNER' |
-	'THANKYOU' |
-	'CANCELLATION';
+// API request models
+export type ReminderPlatform = 'WEB' | 'AMP' | 'MMA' | 'SUPPORT';
```
Comment (TomPretty) [inline src/create-reminder-signup/lambda/models.ts:19]: 👍

File: src/create-reminder-signup/lambda/models.ts [hunk 4]
```diff
@@ -1,25 +1,44 @@
...
 
-export type ReminderStage =
-	'PRE' |
-	'POST' |
-	'WINBACK';
+export type ReminderComponent = 'EPIC' | 'BANNER' | 'THANKYOU' | 'CANCELLATION';
```
Comment (TomPretty) [inline src/create-reminder-signup/lambda/models.ts:21]: 👍

**Review by TomPretty — APPROVED** [review]
Looks good to me! 👍

**Pull request merged**